documentation/src/main/asciidoc/user_guide/locking.adoc

```diff
@@ -10,9 +10,16 @@ MVCC offers many advantages over coarse-grained Java synchronization and even JD
 === Locking implementation details
 Infinispan's MVCC implementation makes use of minimal locks and synchronizations, leaning heavily towards lock-free techniques such as link:http://en.wikipedia.org/wiki/Compare-and-swap[compare-and-swap] and lock-free data structures wherever possible, which helps optimize for multi-CPU and multi-core environments. 
 
-In particular, Infinispan's MVCC implementation is heavily optimized for readers.  Reader threads do not acquire explicit locks for entries, and instead directly read the entry in question.
+In particular, Infinispan's MVCC implementation is heavily optimized for readers.
```

Comment:
IMO, yes. We copy the value to the invocation context and the operation works on that copy.